cart-service/src/order/order.controller.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Controller,
+  Get,
+  Delete,
+  Put,
+  Body,
+  Req,
+  UseGuards,
+  HttpStatus,
+  Param,
+  HttpException,
+} from '@nestjs/common';
+import { OrderService } from './order.service';
+import { BasicAuthGuard } from '../auth';
+import { AppRequest, getUserIdFromRequest } from '../shared';
+import { OrderPayload } from './order.models';
+import { CartService } from '../cart';
+
+@Controller('api/profile/order')
+export class OrderController {
+  constructor(
+    private orderService: OrderService,
+    private cartService: CartService,
+  ) {}
+
+  @UseGuards(BasicAuthGuard)
+  @Put()
+  async create(@Req() req: AppRequest, @Body() body) {
+    const userId = getUserIdFromRequest(req);
+    const { address, items } = body as OrderPayload;
+
+    if (!address || !items) {
```

Comment:
I'm not sure, but it seems that 'items' property is not used while creating an order

Reply:
Yes. It looks you are right. and in DB we don't use it. I will remove it from validation and from TS interfase.